floor_generator/lib/processor/view_processor.dart

```diff
@@ -46,7 +46,24 @@ class ViewProcessor extends QueryableProcessor<View> {
         .getField(AnnotationField.viewQuery)
         ?.toStringValue();
 
-    if (query == null || !query.trimLeft().toLowerCase().startsWith('select')) {
+    if (query == null || !(query.isSelectQuery || query.isCteWithSelect)) {
+      throw _processorError.missingQuery;
+    }
+    return query;
+  }
+}
+
+extension on String {
+  bool get isSelectQuery => toLowerCase().trimLeft().startsWith('select');
+
+  /// whether the string is a common table expression
+  /// followed by a `SELECT` query
+  bool get isCteWithSelect {
+    final lowerCasedString = toLowerCase();
+    return lowerCasedString.trimLeft().startsWith('with') &&
+        'select'.allMatches(lowerCasedString).length >= 2;
+  }
+}
       throw _processorError.missingQuery;
```

Comment:
💡 Oops. It seems like we messed up the file a bit. We can delete this and all following lines.